core/modules/wpublish/QueriesAndChunks.h

```diff
@@ -73,6 +74,7 @@ class QueryStatistics {
     int _tasksCompleted{0};
     int _tasksRunning{0};
     int _tasksBooted{0}; ///< Number of Tasks booted for being too slow.
+    std::atomic<bool> _queryBooted{false}; ///< True when the entire query booted.
```

Comment:
Does it need to be atomic? (My brain only runs single thread today :repeat_one:)